browser/mapping_test.go

```diff
@@ -193,6 +193,91 @@ func TestMappings(t *testing.T) {
 	}
 }
 
+type mockBrowserType struct {
+	api.BrowserType
+
+	launchCalled  bool
+	connectCalled bool
+}
+
+type mockBrowser struct {
+	api.Browser
+}
+
+func (m *mockBrowserType) Launch(opts goja.Value) (_ api.Browser, browserProcessID int) {
+	m.launchCalled = true
+	return &mockBrowser{}, 0
+}
+
+func (m *mockBrowserType) Connect(wsEndpoint string, opts goja.Value) api.Browser {
+	m.connectCalled = true
+	return &mockBrowser{}
+}
+
+func TestBrowserTypeLaunchToConnect(t *testing.T) {
+	t.Parallel()
+
+	type fields struct {
+		vu              moduleVU
+		bt              *mockBrowserType
+		wsURL           string
+		isRemoteBrowser bool
+	}
+
+	mVU := moduleVU{
+		VU: &k6modulestest.VU{
+			RuntimeField: goja.New(),
+		},
+		pidRegistry: &pidRegistry{},
+	}
+
+	testCases := []struct {
+		name             string
+		fields           fields
+		expLaunchCalled  bool
+		expConnectCalled bool
+	}{
+		{
+			name: "should transition from launch to connect",
+			fields: fields{
+				vu:              mVU,
+				bt:              &mockBrowserType{},
+				wsURL:           "wsURL",
+				isRemoteBrowser: true,
+			},
+			expConnectCalled: true,
+		},
+		{
+			name: "should NOT transition from launch to connect",
+			fields: fields{
+				vu: mVU,
+				bt: &mockBrowserType{},
+			},
+			expLaunchCalled: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			launch := mapBrowserType(
+				mVU, tc.fields.bt, tc.fields.wsURL, tc.fields.isRemoteBrowser,
+			)["launch"]
+			launchFn, ok := launch.(func(goja.Value) *goja.Object)
+			if !ok {
```

Comment:
Could you use `require.True` instead?

Reply:
Agree, that's better! :+1: 
Waiting to change this because based on this [discussion](https://github.com/grafana/xk6-browser/pull/819#discussion_r1151840768) we might remove the test completely.

Reply:
Because the mentioned [discussion](https://github.com/grafana/xk6-browser/pull/819#discussion_r1151840768) was resolved, let me know here @ankur22  if you agree on removing the unit test now that we have a working integration test, or if we should address this comment on its implementation :+1:

Reply:
Done :+1: See the current commit for the mapping layer modifications: [926f986](https://github.com/grafana/xk6-browser/pull/819/commits/926f986f88c2a83c941b0c8ad4d0027b75b9fc11).